cdxev/list_command.py

```diff
@@ -0,0 +1,313 @@
+            software_information["name"] = metadata_component.name
+
+        if metadata_component.licenses is not None:
+            license_list = extract_license_strings_from_licenses(
+                metadata_component.licenses
+            )
+            software_information["licenses"] = license_list
+
+        if metadata_component.copyright is not None:
+            software_information["copyright"] = metadata_component.copyright
+
+    else:
+        logger.info(
+            LogMessage(
+                message="SBOM has no metadata.component",
+                description=(
+                    "The SBOM has no metadata so no information "
+                    "about the software could be extracted"
+                ),
+            )
+        )
+
+    return software_information
+
+
+def extract_components_metadata_information(
+    components: list[Component],
+) -> list[dict[str, Any]]:
+    extracted_components = extract_cyclonedx_components(components)
+
+    list_of_license_information: list[dict[str, Any]] = []
+
+    for component in extracted_components:
+        license_information: dict[str, Any] = {}
+        not_empty = False
+        if component.name is not None:
+            license_information["name"] = component.name
+
+        if component.licenses is not None:
+            license_list = extract_license_strings_from_licenses(component.licenses)
+            license_information["licenses"] = license_list
+            not_empty = True
+        if component.copyright is not None:
+            license_information["copyright"] = component.copyright
+            not_empty = True
+
+        if not_empty:
+            list_of_license_information.append(license_information)
+
+    return list_of_license_information
+
+
+def write_list_to_str(str_list: list[str], division_character: str = "\n") -> str:
+    string = ""
+    if str_list:
+        string += str_list[0]
+    for index in range(1, len(str_list)):
+        string += division_character
+        string += str_list[index]
+    return string
+
+
+def write_license_dict_to_txt(info_dict: dict) -> str:
+    string = ""
+
+    if info_dict.get("name", ""):
+        string += info_dict.get("name", "") + ":"
+
+    if info_dict.get("copyright", ""):
+        string += "\n"
+        string += info_dict.get("copyright", "")
+
+    if info_dict.get("licenses", ""):
+        license_str = write_list_to_str(info_dict.get("licenses", ""))
+        string += "\n"
+        string += license_str
+
+    if not info_dict.get("licenses", "") and not info_dict.get("copyright", ""):
+        string += "\n"
+        string += "No license or copyright information available"
+
+    return string
+
+
+def write_license_dict_to_csv(info_dict: dict) -> str:
+    string = ""
+
+    if info_dict.get("name", ""):
+        string += info_dict.get("name", "")
+
+    if info_dict.get("copyright", ""):
+        string += ","
+        string += info_dict.get("copyright", "")
+
```

Comment:
this leads to a malformed csv, because if there is sometimes copyright and sometimes not, you get:

```json
Name,Copyright,License
comp1,copy1,lic1
comp2,lic2
```
Which is not allowed, as each row should have the same amount of columns.

Reply:
added the "" arounf the entries, thwe licenses are now seperated with a ; 

"name","copyright","license_1;license_2;license_3"
 so , can be used as seperator for the csv document and ; as seperator for the licenses list